src/test/datascience/notebook/intellisense/completionProvider.vscode.test.ts

```diff
@@ -0,0 +1,92 @@
+    waitForKernelToGetAutoSelected,
+    prewarmNotebooks
+} from '../helper';
+
+/* eslint-disable @typescript-eslint/no-explicit-any, no-invalid-this */
+suite('DataScience - VSCode Notebook - (Code Completion via Jupyter) (slow)', function () {
+    let api: IExtensionTestApi;
+    let editorProvider: INotebookEditorProvider;
+    const disposables: IDisposable[] = [];
+    let vscodeNotebook: IVSCodeNotebook;
+    let completionProvider: NotebookCompletionProvider;
+    this.timeout(120_000);
+    suiteSetup(async function () {
+        this.timeout(120_000);
+        api = await initialize();
+        if (!(await canRunNotebookTests())) {
+            return this.skip();
+        }
+        await trustAllNotebooks();
+        await startJupyterServer();
+        await prewarmNotebooks();
+        sinon.restore();
+        vscodeNotebook = api.serviceContainer.get<IVSCodeNotebook>(IVSCodeNotebook);
+        editorProvider = api.serviceContainer.get<INotebookEditorProvider>(INotebookEditorProvider);
+        completionProvider = api.serviceContainer.get<NotebookCompletionProvider>(NotebookCompletionProvider);
+    });
+    // Use same notebook without starting kernel in every single test (use one for whole suite).
+    setup(async function () {
+        traceInfo(`Start Test ${this.currentTest?.title}`);
+        sinon.restore();
+        await startJupyterServer();
+        // Open a notebook and use this for all tests in this test suite.
+        await editorProvider.createNew();
+        await waitForKernelToGetAutoSelected(undefined);
+        await deleteAllCellsAndWait();
+        assert.isOk(vscodeNotebook.activeNotebookEditor, 'No active notebook');
+        traceInfo(`Start Test (completed) ${this.currentTest?.title}`);
+    });
+    teardown(async function () {
+        traceInfo(`Ended Test ${this.currentTest?.title}`);
+        await closeNotebooksAndCleanUpAfterTests(disposables);
+        traceInfo(`Ended Test (completed) ${this.currentTest?.title}`);
+    });
+    suiteTeardown(() => closeNotebooksAndCleanUpAfterTests(disposables));
+    test('Execute cell and get completions for variable', async () => {
+        await insertCodeCell('a = 1', { index: 0 });
+        const cell = vscodeNotebook.activeNotebookEditor?.document.cells![0]!;
+
+        await executeCell(cell);
+
+        // Wait till execution count changes and status is success.
+        await waitForExecutionCompletedSuccessfully(cell);
+
+        await insertCodeCell('a.', { index: 1 });
+        const cell2 = vscodeNotebook.activeNotebookEditor!.document.cells[1];
+
+        const token = new CancellationTokenSource().token;
+        const position = new Position(0, 2);
+        const context: CompletionContext = {
+            triggerKind: CompletionTriggerKind.TriggerCharacter,
+            triggerCharacter: '.'
+        };
+        const completions = await completionProvider.provideCompletionItems(cell2.document, position, token, context);
```

Comment:
This is the same completion provider we hook up with VS Code, i.e. VSCode will use this API.
Hence testing this is sufficient (also this is how VS Code recommends testing such code).
Unless we do full blown UI test where we launch an editor & click it etc, which feels unnecessary as that's VSC codes job and we're provide the API they require.